01_easy_algorithms/05_count_lone_ones/05_count_lone_ones.rb

```diff
@@ -0,0 +1,17 @@
+def count_lone_ones(x)
+  arr1 = []
+  arr_x = x.digits.reverse
+  z = arr_x.length
+  idx=0
+  while idx < z
+    if arr_x[idx] == 1 && arr_x[idx + 1] == 1
+      then idx += 1
+    elsif arr_x[idx] == 1
+      arr1 << 1  
+    else
+      arr1 << 3  
+    end  
+    idx += 1
+  end
+  arr1.count(1)  
+end
```

Comment:
The code works but can be improved. You are adding 1 to arr1 when there is a lone one and adding 3 when there isn't a lone one. As a concept, that isn't the natural way a human would solve this problem. If another developer comes along and looks at your code, they'll wonder why you're adding three and not 2 or 4 etc.

Also instead of adding elements to the array and returning the count of it, just make a variable (count = 0) and keep adding to that when you come across a lone one.